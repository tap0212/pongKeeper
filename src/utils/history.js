import { createBrowserHistory } from 'history';
const baseUrl = process.env.NODE_ENV === 'production' ? '/pong-keeper' : '/';
const history = createBrowserHistory({ basename: baseUrl });
export default history;
