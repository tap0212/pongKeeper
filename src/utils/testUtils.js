import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import configureStore from '../configureStore';

export const renderProvider = (children) => {
  const store = configureStore({}, browserHistory).store;
  return render(
    <Provider store={store}>
      <ThemeProvider
        theme={{
          main: 'violet'
        }}
      >
        {children}
      </ThemeProvider>
    </Provider>
  );
};
