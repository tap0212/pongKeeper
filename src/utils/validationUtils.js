// regex validation rules
const USERNAME_VALIDATION_RULE = /^[a-zA-Z0-9]+$/;
// method which creates rules for antd input fields
const createRule = (ruleName, ruleValue, message) => ({
  [ruleName]: ruleValue,
  message
});
// field is required?
export const required = (message) => createRule('required', true, message);
const pattern = (pattern, message) => createRule('pattern', pattern, message);

export const isUserNameValid = (requiredMsgId, invalidFullnameMsgId) => [
  required(requiredMsgId),
  pattern(USERNAME_VALIDATION_RULE, invalidFullnameMsgId)
];
