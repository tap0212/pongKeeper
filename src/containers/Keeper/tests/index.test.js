/**
 *
 * Tests for Keeper
 *
 *
 */

import React from 'react';
import { renderProvider } from '../../../utils/testUtils';
import { KeeperTest as Keeper } from '../index';

describe('<Keeper /> container tests', () => {
  beforeEach(() => {});
  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(<Keeper />);
    expect(baseElement).toMatchSnapshot();
  });
});
