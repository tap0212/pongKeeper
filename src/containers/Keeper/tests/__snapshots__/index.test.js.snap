// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Keeper /> container tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-dlfnbm bFbQkn"
    >
      <div
        class="sc-hKgILt RrfSS"
      >
        <p
          class="sc-gKsewC iSQBEZ"
        >
          The Keeper
        </p>
      </div>
    </div>
  </div>
</body>
`;
